src/App.tsx

```diff
@@ -1,93 +1,47 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useMemo } from 'react';
+import { useLocation } from 'react-router-dom';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { filterTodos } from './utils/filterTodos';
+import { useLocaleStorage } from './utils/useLocaleStorage';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useLocaleStorage('todos');
+  const amountCompletedTodos = useMemo(() => {
+    return todos.reduce((acc, curr) => {
+      return curr.completed ? acc + 1 : acc;
+    }, 0);
+  }, [todos]);
+  const { pathname: location } = useLocation();
+  const isCompletedTodo = useMemo(() => {
+    return !todos.some(todo => todo.completed === false);
+  }, [todos]);
```

Comment:
Probably my code doesn't work, but anyway your can simplify your code:
```suggestion
  const hasCompletedTodo = useMemo(() => todos.some(todo => todo.completed), [todos]);
```